pkg/kubelet/prober/worker.go

```diff
@@ -107,14 +112,24 @@ func newWorker(
 		w.initialValue = results.Unknown
 	}
 
+	podNameSlice := strings.Split(w.pod.Name, "-")
```

Comment:
@dgrisonnet is this what we had in mind?

I can clean up the implementation after we agree on an approach (is there a common lib that will do this pod UID redaction in a more resilient way?), write UT, etc.

cc @SergeyKanzhelev

Reply:
Yeah, that's similar to what I had in mind but I don't think that will cover all the use-cases. We have a few possible scenarios here:
1. static pods / pods created manually
2. pods created by a replicaset/statefulset
3. pods created by a deployment

Scenario 2 can be handled by relying on pod.GenerateName which contains pod.Name + "-".

For scenario 3 it is a bit more complex because the resulting pod will be named pod.Name + pod template hash + uid and pod.GenerateName will be equal to pod.Name + "-" + pod template hash + "-". We can rely on pod.Labels["pod-template-hash"] to get the pod template hash and then deduce the pod name without UIDs.

As for scenario 1, we don't want to do anything since they won't have UIDs. We just have to make sure that it is still possible to have names with hyphens and without UIDs.

So my suggestion would to have something like:
```go
podName = w.pod.Name
if w.pod.GenerateName != "" {
  podNameSlice := strings.Split(w.pod.Name, "-")
  podName = strings.Join(podNameSlice[:len(podNameSlice)-1], "-")
  if label, ok := w.pod.GetLabels()[apps.DefaultDeploymentUniqueLabelKey]; ok {
    podName = strings.ReplaceAll(podName, fmt.Sprintf("-%s", label), "")
  }
}
```
It is pretty rough, so not sure if that would be the best option to go with, but it should give you a better idea of the different scenarios that we need to support here.

Reply:
@dgrisonnet 
can we unhold this pr now?